Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/PolDiffILLReduction.py

```diff
@@ -0,0 +1,876 @@
+            tmp_name = '{0}_{1}_{2}'.format(ws[2:], mtd[ws_00].getRun().getLogData('POL.actual_state').value, index)
+            Divide(LHSWorkspace=mtd[ws_00]-mtd[ws_01],
+                   RHSWorkspace=(2*flipper_eff-1)*mtd[ws_00]+mtd[ws_01],
+                   OutputWorkspace=tmp_name)
+            tmp_names.append(tmp_name)
+            if self._method_data_structure == 'Uniaxial' and entry_no % 2 == 1:
+                index += 1
+            elif self._method_data_structure == 'XYZ' and entry_no % 6 == 5:
+                index += 1
+            elif self._method_data_structure == '10p' and entry_no % 10 == 9:
+                index += 1
+        GroupWorkspaces(InputWorkspaces=tmp_names, OutputWorkspace='tmp')
+        DeleteWorkspaces(ws)
+        RenameWorkspace(InputWorkspace='tmp', OutputWorkspace=ws)
+        return ws
+
+    def _detector_analyser_energy_efficiency(self, ws):
+        """Corrects for the detector and analyser energy efficiency."""
+        correction_formula = "1.0 - exp(-13.153/sqrt(e))"
+        for entry in mtd[ws]:
+            SetInstrumentParameter(entry, ParameterName="formula_eff", Value=correction_formula)
+            DetectorEfficiencyCorUser(InputWorkspace=entry, OutputWorkspace=entry,
+                                      IncidentEnergy=self._experimentProperties['initial_energy'].value)
+        return ws
+
+    def _frame_overlap_correction(self, ws):
+        pass
+
+    def _apply_polarization_corrections(self, ws, pol_eff_ws):
+        """Applies the polarisation correction based on the output from quartz reduction."""
+        fp = 1 # flipper efficiency, placeholder
+        nPolarisations = None
+        singleCorrectionPerPOL = False
+        if mtd[ws].getNumberOfEntries() != 2*mtd[pol_eff_ws].getNumberOfEntries():
+            singleCorrectionPerPOL = True
+            if self._method_data_structure == 'Uniaxial':
+                nPolarisations = 2
+            elif self._method_data_structure == 'XYZ':
+                nPolarisations = 3
+            elif self._method_data_structure == '10p':
+                nPolarisations = 5
+            if mtd[pol_eff_ws].getNumberOfEntries() != nPolarisations:
+                raise RuntimeError("Incompatible number of polarisations between quartz input and sample.")
+
+        for entry_no in range(mtd[ws].getNumberOfEntries()):
+            if entry_no % 2 != 0:
+                continue
+            polarisation_entry_no = int(entry_no/2)
+            if singleCorrectionPerPOL:
+                polarisation_entry_no = int(polarisation_entry_no % nPolarisations)
+            phi = mtd[pol_eff_ws].getItem(polarisation_entry_no).name()
+            intensity_0 = mtd[ws].getItem(entry_no).name()
+            intensity_1 = mtd[ws].getItem(entry_no+1).name()
+            tmp_names = [intensity_0 + '_tmp', intensity_1 + '_tmp']
+
+            Divide(LHSWorkspace=((1.0-mtd[phi])*(1.0-fp) + fp*(1+mtd[phi]))*mtd[intensity_0]
```

Comment:
This is not at all readable. Why can we not reuse the existing algorithms to derive/perform polarisation corrections?

Reply:
This has changed to be more readable. We agreed to not use the existing algorithms as they require more intermediate steps and inputs while the calculation for D7 is currently very simple. Let me know if something changed about this decision.